src/pages/library/common.js

```diff
@@ -192,8 +195,32 @@ export const selectionActionComponent = (selectedData, setSelectedData) => {
           }, 'Cancel'),
           h(ButtonPrimary, {
             style: { textTransform: 'none', fontSize: 14 },
-            onClick: () => {}
+            onClick: () => {
+              setShowExportModal(true)
+            }
           }, ['Save to a workspace'])
+        ]),
+        showExportModal &&
+        h(Modal, {
+          title: 'Save to a Terra workspace',
+          onDismiss: () => setShowExportModal(false),
+          okButton: () => {
+            Nav.history.push({
+              pathname: Nav.getPath('import-data'),
+              search: `?url=https://jade.datarepo-dev.broadinstitute.org/&snapshotId=idid&snapshotName=${datasetName}&format=snapshot`
+            })
+          }
+        }, [
+          div([
+            div('Data with this name will be saved in Terra. If this name already exists, it will be overwritten.', []),
+            div({ style: { marginTop: '1rem' } }, [
+              h(TextInput, {
+                placeholder: 'Name of data',
+                value: datasetName,
+                onChange: setDatasetName
+              })
+            ])
+          ])
```

Comment:
The diff below is to show some simplifications:

```suggestion
          'Data with this name will be saved in Terra. If this name already exists, it will be overwritten.',
          div({ style: { marginTop: '1rem' } }, [
            h(TextInput, {
              placeholder: 'Name of data',
              value: datasetName,
              onChange: setDatasetName
            })
          ])
```

Also, we require `TextInput` to have a `label` for accessibility. Please see other instances in the code for examples of it.